torch/linalg/__init__.py

```diff
@@ -214,6 +214,86 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+cond = _add_docstr(_linalg.linalg_cond, r"""
+linalg.norm(input, p=None, *, out=None) -> Tensor
+
+Computes the condition number of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`, using the matrix norm defined by :attr:`p`.
+The condition number is defined as the matrix norm of
+:attr:`input` times the matrix norm of the inverse of :attr:`input`.
+
+This function supports ``float``, ``double``, and only on CPU, ``cfloat`` and ``cdouble`` dtype for :attr:`input`.
+
+.. note:: For `p = {None, 2, -2}` :attr:`input` can be non-square. For other norm types :attr:`input` must be
+          a square matrix or a batch of square matrices. If :attr:`input` does not satisfy this requirement
+          then a RuntimeError will be thrown.
+
+.. note:: If :attr:`input` is a non-invertible matrix then a tensor containing infinity will be returned.
+          If :attr:`input` is a batch of matrices and one or more of them is not invertible
+          then a RuntimeError will be thrown.
+
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+
+Args:
+    input (Tensor): the input matrix of size :math:`(m, n)` or the batch of matrices of size :math:`(*, m, n)`
+                    where `*` is one or more batch dimensions.
+
+    p (int, float, inf, -inf, 'fro', 'nuc', optional): the type of the matrix norm to use in the computations.
+        The following norms are supported:
+
+        =====  ============================
+        p      norm for matrices
+        =====  ============================
+        None   2-norm (ratio of the largest singular value to the smallest singular value)
+        'fro'  Frobenius norm
+        'nuc'  nuclear norm
+        inf    max(sum(abs(x), dim=1))
+        -inf   min(sum(abs(x), dim=1))
+        1      max(sum(abs(x), dim=0))
+        -1     min(sum(abs(x), dim=0))
+        2      2-norm (ratio of the largest singular value to the smallest singular value)
+        -2     ratio of the smallest singular value to the largest singular value
```

Comment:
It seems like the line above should just be the phrase in the parentheses to be consistent with the presentation of the -2 norm and the other norms. That is,

```
2   ratio of the largest singular value to the smallest singular value
-2 ratio of the smallest singular value to the largest singular value
```